src/legacy/views/secondary-bar/parts/edit/share-folder-properties.tsx

```diff
@@ -145,35 +162,59 @@ const Grantee = ({
 				onMouseLeave={onMouseLeave}
 				onMouseEnter={onMouseEnter}
 				grant={grant}
-				createSnackbar={createSnackbar}
-				folders={folders}
-				allCalendars={allCalendars}
-				setModal={setModal}
 				setActiveModal={setActiveModal}
-				totalAppointments={totalAppointments}
-				setActiveGrant={setActiveGrant}
 			/>
 		</Container>
 	);
 };
 
 export const ShareFolderProperties = ({
 	folder,
-	createSnackbar,
-	folders,
-	allCalendars,
-	setModal,
-	setActiveGrant,
-	totalAppointments,
 	setActiveModal
-}) => {
+}: ShareFolderPropertiesProps): React.JSX.Element => {
+	const createSnackbar = useSnackbar();
 	const [t] = useTranslation();
-	const { grant } = folder.sharedWith;
+	const [grant, setGrant] = useState<Array<Grant> | undefined>();
 	const shareFolderRoleOptions = useMemo(
-		() => ShareFolderRoleOptions(t, grant.perm?.includes('p')),
-		[t, grant.perm]
+		() => ShareFolderRoleOptions(t, grant?.[0]?.perm?.includes('p')),
+		[t, grant]
 	);
 
+	useEffect(() => {
+		soapFetch<GetFolderActionRequest, GetFolderActionResponse | ErrorSoapBodyResponse>(
+			'GetFolder',
+			{
+				_jsns: 'urn:zimbraMail',
+				folder: { l: folder.id }
+			}
+		)
+			.then((response): void => {
+				if ('Fault' in response) {
+					createSnackbar({
+						key: new Date().toDateString(),
+						replace: true,
+						type: 'error',
+						label: t('label.error_try_again', 'Something went wrong, please try again'),
+						autoHideTimeout: 3000,
+						hideButton: true
+					});
+					throw new Error(response.Fault.Reason.Text, { cause: response.Fault });
+				}
+				if (response && response?.folder) {
+					setGrant(response.folder[0].acl.grant);
+				}
+			})
+			.catch(() => {
+				createSnackbar({
+					key: new Date().toDateString(),
+					replace: true,
+					type: 'error',
+					label: t('label.error_try_again', 'Something went wrong, please try again'),
+					autoHideTimeout: 3000,
+					hideButton: true
+				});
+			});
```

Comment:
Isn't the snackbar inside the then block redundant, since the error is also explicitly thrown? The thrown error should be catched inside the following catch block